core/server/data/validation/index.js

```diff
@@ -42,6 +43,27 @@ validator.extend('isTimezone', function isTimezone(str) {
     return moment.tz.zone(str) ? true : false;
 });
 
+validator.extend('isEmptyOrWebsite', function isEmptyOrWebsite(str) {
```

Comment:
> Are we sure query strings aren't valid for a website address? Eg. http://somesocialsite.com/?user=myusername is perfectly valid IMO. I think the thing we need to protect against is injected scripts which will mostly include < and > characters.

This is valid, but I don't see any reason why we should support this case right now, without knowing if people want or need it. I'd suggest we keep this as simple as is and merge it 🐩